analyzer/codechecker_analyzer/analyzers/clangsa/ctu_autodetection.py

```diff
@@ -0,0 +1,134 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+"""
+Clang Static Analyzer related functions.
+"""
+
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+import os
+import re
+import subprocess
+
+from codechecker_common.logger import get_logger
+
+LOG = get_logger('analyzer.clangsa')
+
+
+def test_binary_existence(tool_path):
+    """ Test the executable for existence by checking its version. """
+    try:
+        version = subprocess.check_output([tool_path, '-version'])
+    except (subprocess.CalledProcessError, OSError):
+        version = 'ERROR'
+    return version != 'ERROR'
+
+
+class CTUAutodetection(object):
+
+    def __init__(self, analyzer_binary):
+        self.__analyzer_binary = analyzer_binary
+        self.__analyzer_version_info = None
+
+    @property
+    def analyzer_version_info(self):
+        """ Returns the relevant parameters of the analyzer by parsing the
+        output of the analyzer binary when called with version flag. """
+
+        if self.__analyzer_binary is None:
+            return False
+
+        if self.__analyzer_version_info is not None:
+            return self.__analyzer_version_info
+
+        try:
+            analyzer_output = subprocess.check_output(
+                [self.__analyzer_binary, '--version'])
+        except (subprocess.CalledProcessError, OSError):
+            return False
+
+        version_pattern = re.compile(
```

Comment:
If I am using an official pre-built LLVM release from http://releases.llvm.org/, the version information will look like this:
```
clang version 7.0.0 (tags/RELEASE_700/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/username/clang+llvm-7.0.0-x86_64-linux-gnu-ubuntu-16.04/bin
```
This regex will not match to this output.

Reply:
Fair point, I will extend the implementation to handle this case as well.